macOS/DuckDuckGoVPN/VPNProxyLauncher.swift

```diff
@@ -47,24 +47,35 @@ final class VPNProxyLauncher {
 
     // MARK: - Status Changes
 
+    private struct ProxyStatusUpdate: Equatable {
+        let newStatus: NEVPNStatus
+        let isProxyStatusChange: Bool
+    }
+
     private func subscribeToStatusChanges() {
         notificationCenter.publisher(for: .NEVPNStatusDidChange)
             .receive(on: DispatchQueue.main)
-            .sink { [weak self] notification in
-                self?.statusChanged(notification: notification)
+            .compactMap { [proxyController] notification in
+                guard let connection = notification.object as? NEVPNConnection else {
+                    return nil
+                }
+
+                return ProxyStatusUpdate(
+                    newStatus: connection.status,
+                    isProxyStatusChange: proxyController.isUsingConnection(connection))
+            }
+            .removeDuplicates()
+            .sink { [weak self] (proxyStatusUpdate: ProxyStatusUpdate) in
+                self?.statusChanged(
+                    newStatus: proxyStatusUpdate.newStatus,
+                    isProxyStatusChange: proxyStatusUpdate.isProxyStatusChange)
             }
             .store(in: &cancellables)
     }
 
-    private func statusChanged(notification: Notification) {
+    private func statusChanged(newStatus: NEVPNStatus, isProxyStatusChange: Bool) {
         Task { @MainActor in
-            guard let connection = notification.object as? NEVPNConnection else {
-                return
-            }
-
-            let isProxyConnectionStatusChange = proxyController.isUsingConnection(connection)
-
-            try await startOrStopProxyIfNeeded(isProxyConnectionStatusChange: isProxyConnectionStatusChange)
+            try await startOrStopProxyIfNeeded(isProxyStatusChange: isProxyStatusChange)
```

Comment:
When this is called, it can trigger another status update with the same status.  This is because loading the VPN from settings pushes a status update.

This results in an infinite loop since we're listening for status updates here.